web_monitoring/internetarchive.py

```diff
@@ -71,9 +71,8 @@ def original_url_for_memento(memento_url):
     >>> original_url_for_memento('http://web.archive.org/web/20170813195036/https://arpa-e.energy.gov/?q=engage/events-workshops')
     'https://arpa-e.energy.gov/?q=engage/events-workshops'
     """
-    try:
-        url = MEMENTO_URL_PATTERN.match(memento_url).group(1)
-    except:
+    url = MEMENTO_URL_PATTERN.match(memento_url).group(1)
+    if url is None:
```

Comment:
Hmmm, this was my lazy way of handling both non-matches (because calling `.group(1)` would raise an `AttributeError`) and bad input types (because calling `.match(not_a_string)` would raise a `TypeError`). Also, I’m not sure this condition can ever be true (if there was a match but nothing in group 1, we’d get `url == ''`).

I think I’m going to change this to split the difference — let the `.match()` call raise a `TypeError` without catching it and recasting as a `ValueError`, then check whether it matched before getting the capture group so we don’t have to have a `try...except` at all.

Seem like a good approach, @danielballan? Otherwise I can flip back to the `try...except` approach but without a *bar* `except`.

Reply:
See 4551a9be7c3daa77a9145134bc1415f9ce616305

Reply:
:+1: